Bitbucket.Authentication/BasicAuth/BasicAuthAuthenticator.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Atlassian.Bitbucket.Authentication.Rest;
+using Microsoft.Alm.Authentication;
+using Microsoft.Alm.Git;
+
+namespace Atlassian.Bitbucket.Authentication.BasicAuth
+{
+    class BasicAuthAuthenticator
+    {
+        public async Task<AuthenticationResult> GetAuthAsync(TargetUri targetUri, TokenScope scope, int requestTimeout, Uri restRootUrl, string username, string password)
+        {
+            // use the provided username and password and attempt a Basic Auth request to a known
+            // REST API resource.
+            
+            string basicAuthValue = String.Format("{0}:{1}", username, password);
+            byte[] authBytes = Encoding.UTF8.GetBytes(basicAuthValue);
+            basicAuthValue = Convert.ToBase64String(authBytes);
+            var authHeader = "Basic " + basicAuthValue;
+
+            var result = await RestClient.TryGetUser(targetUri, requestTimeout, restRootUrl, authHeader);
+
+            if (result.Type.Equals(AuthenticationResultType.Success))
+            {
+                // Success with username/passord indicates 2FA is not on so
+                // the 'token' is actually the password if we had a
+                // successful call then the password is good.
+                var token = new Token(password, TokenType.Personal);
+                if(!string.IsNullOrWhiteSpace(result.RemoteUsername) && !username.Equals(result.RemoteUsername))
+                {
+                    Trace.WriteLine($"Remote username [{result.RemoteUsername}] != [{username}] supplied username");
+                    return new AuthenticationResult(AuthenticationResultType.Success, token, result.RemoteUsername);
+                }
```

Comment:
If the username stored in Bitbucket isn't the one supplied by the user, e.g. they used an email address, then keep hold of the Bitbucket one because we will store that for the credentials